src/DDTrace/Configuration/EnvVariableRegistry.php

```diff
@@ -102,6 +102,49 @@ public function floatValue($key, $default, $min = null, $max = null)
         return $this->registry[$key];
     }
 
+    /**
+     * Given a string like 'key1:value1,key2:value2', it returns an associative array
+     * ['key1'=> 'value1', 'key2'=> 'value2']
+     *
+     * @param string $key
+     * @return string[]
+     */
+    public function associativeStringArrayValue($key)
+    {
+        if (isset($this->registry[$key])) {
+            return $this->registry[$key];
+        }
+
+        $default = [];
+        $value = self::get($key);
+
+        if (null === $value) {
+            return $this->registry[$key] = $default;
```

Comment:
My bad for not catching this in the first CR! But this shouldn't save the default value:
```suggestion
            return $default;
```
Otherwise we'll get some weird behavior like this:
```php
var_dump($config->getString('some_unset_val', 'foo')); // "foo"
// ...and then later on...
var_dump($config->getString('some_unset_val', 'bar')); // "foo" (still)
```
I totally missed that in the first CR! Doh! :D

Reply:
Ohh nice catch!!!